plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -579,16 +584,90 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return Optional.of(new ConstraintApplicationResult<>(handle, TupleDomain.withColumnDomains(unsupported), newExpression, false));
     }
 
-    protected static String likeToRegexp(String like)
+    protected static boolean isSupportedLikeCall(Call call)
+    {
+        if (!LIKE_PATTERN_FUNCTION_NAME.equals(call.getFunctionName())) {
+            return false;
+        }
+
+        List<ConnectorExpression> arguments = call.getArguments();
+        if (arguments.size() < 2 || arguments.size() > 3) {
+            return false;
+        }
+
+        if (!(arguments.get(0) instanceof Variable) || !(arguments.get(1) instanceof Constant)) {
+            return false;
+        }
+
+        if (arguments.size() == 3) {
+            return arguments.get(2) instanceof Constant;
+        }
+
+        return true;
+    }
+
+    protected static String likeToRegexp(Slice pattern, Optional<Slice> escape)
+    {
+        char escapeChar = escape.map(ElasticsearchMetadata::getEscapeChar)
+                //
+                .orElse('\0');
```

Comment:
`(char) 0` is not a reserved character. It's a legal character in Java.
Add a test showing that `likeToRegexp("\000%", Optional.empty())` returns "\000.*".